airflow/decorators/base.py

```diff
@@ -207,17 +209,29 @@ def __init__(
         super().__init__(task_id=task_id, **kwargs_to_upstream, **kwargs)
 
     def execute(self, context: Context):
+        # todo make this more generic (move to prepare_lineage) so it deals with non taskflow operators
+        #  as well
+        for arg in chain(self.op_args, self.op_kwargs.values()):
+            if isinstance(arg, Dataset):
+                self.inlets.append(arg)
         return_value = super().execute(context)
         return self._handle_output(return_value=return_value, context=context, xcom_push=self.xcom_push)
 
     def _handle_output(self, return_value: Any, context: Context, xcom_push: Callable):
         """
         Handles logic for whether a decorator needs to push a single return value or multiple return values.
+        It sets outlets if any datasets are found in the returned value(s)
```

Comment:
Doc style nit. The first paragraph appears in the summary so there should be a blank line. Doesn't matter here as this is a private method so wouldn't appear in docs anyway, but a good habit to get in to.
```suggestion

        It sets outlets if any datasets are found in the returned value(s)
```